src/Compilers/CSharp/Test/Emit/Attributes/AttributeTests_NativeInteger.cs

```diff
@@ -1022,6 +1022,52 @@ void assert(string expectedType, string fieldName)
             }
         }
 
+        [Fact]
+        public void FunctionPointersWithNativeIntegerTypes()
+        {
+            var comp = CompileAndVerify(@"
+unsafe class C
+{
+    public delegate*<nint, nint, nint> F1;
+    public delegate*<System.IntPtr, System.IntPtr, nint> F2;
+    public delegate*<nint, System.IntPtr, System.IntPtr> F3;
+    public delegate*<System.IntPtr, nint, System.IntPtr> F4;
+    public delegate*<delegate*<System.IntPtr, System.IntPtr, System.IntPtr>, nint> F5;
+    public delegate*<nint, delegate*<System.IntPtr, System.IntPtr, System.IntPtr>> F6;
+}
+", options: TestOptions.UnsafeReleaseDll, parseOptions: TestOptions.RegularPreview, symbolValidator: symbolValidator);
+
+            void symbolValidator(ModuleSymbol module)
+            {
+                var expectedAttributes = @"
+C
+    [NativeInteger({ False, True, True, True })] delegate*<System.IntPtr, System.IntPtr, System.IntPtr> F1
```

Comment:
>False, True, True, True [](start = 21, length = 23)

Filed [https://github.com/dotnet/roslyn/issues/44086](https://github.com/dotnet/roslyn/issues/44086) for native int encoding question. Not sure why we would want to encode a bool for a type that is not an U/IntPtr.